packages/mobile/src/fiatExchanges/ProviderOptionsScreen.tsx

```diff
@@ -120,7 +125,8 @@ function ProviderOptionsScreen({ route, navigation }: Props) {
 
   const asyncUserLocation = useAsync(async () => {
     try {
-      const { alpha2, state } = await fetchLocationFromIpAddress()
+      const { alpha2, state, ipAddress } = await fetchLocationFromIpAddress()
+      setUserIpAddress(ipAddress)
```

Comment:
Since this is inside a `useAsync` this setter could be called after the component unmounts. Instead, we should just return the `ipAddress` along with the `country` and `state` and access it calling `.result` like we do for those other fields.